plugins/wasm_bpf/wasm-bpf.cpp

```diff
@@ -0,0 +1,243 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2019-2022 Second State INC
+
+#include <asm/unistd.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <string>
+#include <unistd.h>
+
+#include "bpf-api.h"
+#include "common/types.h"
+#include "wasmedge/wasmedge.h"
+
+extern "C" {
+#include <bpf/bpf.h>
+#include <bpf/libbpf.h>
+}
+
+static int32_t bpf_buffer_sample(void *ctx, void *data, size_t size);
+static int32_t libbpf_print_fn(enum libbpf_print_level level,
+                               const char *format, va_list args) {
+  if (level == LIBBPF_DEBUG && DEBUG_LIBBPF_RUNTIME)
+    return 0;
+  char buf[DEBUG_PRINT_BUFFER_SIZE];
+  int32_t len = vsnprintf(buf, sizeof(buf), format, args);
+  spdlog::debug("[WasmEdge Wasm_bpf] {}", buf);
```

Comment:
Maybe it's not the best way to do it?

Reply:
Seems there is no other ways.. `libbpf_print_fn` accepts a format string and a va_list. But spdlog seems not have functions that provide such ability..